app/js/diceware.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * diceware.js - generate N distinct words from a dictionary of 7776 possibilities
+ */
+
+function Diceware() {
+  this.wordlist = [];
+  if (typeof Math.log2 === 'undefined') {
+    Math.log2 = function(N) {
+        return Math.log(N)/Math.log(2) 
+    };
+  }
+}
+
+Diceware.prototype.load = function(callback) {
+  var that = this;
+  $.get("/data/diceware.wordlist.asc", function(data) {
+    var line = '';
+    var split = data.split("\n");
+    var l = split.length;
+    for (var i = 0; i < l; i++) {
+      line = split[i];
+      if (line === '-----BEGIN PGP SIGNED MESSAGE-----' || line === '') {
+          continue;
+      }
+      if (line === '-----BEGIN PGP SIGNATURE-----') {
+          break;
+      }
+      var myregexp = /^\d+\s*([^\s]+)$/;
+      var match = myregexp.exec(line);
+      if (match != null) {
+        // matched text: match[0]
+        // match start: match.index
+        // capturing group n: match[n]
+        that.wordlist.push(match[1]);
+      }
+    }
+    callback();
+  });
+}
+
+/**
+ * Return a number of diceware words
+ * @param int words - the number of words to generate
+ */
+Diceware.prototype.getWords = function(words) {
+  if (!window.crypto || !window.crypto.getRandomValues) {
+    return [];
+  }
+  var diced = [];
+  
+  for (var i = 0; i < words; ++i) {
+    diced.push(this.getSingleWord());
+  }
+  return diced;
+};
+
+// http://stackoverflow.com/questions/18230217/javascript-generate-a-random-number-within-a-range-using-crypto-getrandomvalues 
+
+Diceware.prototype.random = function (min, max) {
```

Comment:
@sarciszewski, this PR was linked from [StackOverflow question](https://stackoverflow.com/questions/18230217/javascript-generate-a-random-number-within-a-range-using-crypto-getrandomvalues) and I suspect many people will reuse this code.
However, function is not working properly and often returns negative numbers out of specified range.
Demo: https://jsfiddle.net/oyd9ptnv/

Reply:
This is because bitwise operations in JS are limited to <32 bits